signac/contrib/collection.py

```diff
@@ -503,7 +522,7 @@ def __setitem__(self, _id, doc, _trust=False):
         if _trust:
             self._docs[_id] = doc
         else:
-            self._docs[_id] = json.loads(json.dumps(doc))
+            self._docs[_id] = self._validate_doc(json.loads(json.dumps(doc)))
```

Comment:
Is the `loads(dumps(doc))` just to ensure that the document is valid JSON? If not, then this is a no-op, right? And if so, should we wrap it in a `try...except` to provide a more informative error?

Reply:
Yes, exactly. I'll see if I can easily provide an informative error message.

Reply:
I forgot to mention, it also ensures type normalization.